cmake/Utils.cmake

```diff
@@ -235,7 +236,6 @@ macro(xgboost_target_link_libraries target)
 
   if(USE_CUDA)
     xgboost_set_cuda_flags(${target})
-    target_link_libraries(${target} PUBLIC CUDA::cudart_static)
```

Comment:
As we are using `PUBLIC` here, am guessing we want to do the same in this change

Reply:
@jakirkham No, setting this `PUBLIC` will adversely affect the downstream users of the C API.

Reply:
Actually, maybe downstream users need to install CUDA runtime too? Let me take another look.

Reply:
Ok happy to be corrected here. Just thought we wanted the same behavior

Reply:
Yeah, another test with the C API demo (https://github.com/dmlc/xgboost/tree/master/demo/c-api/basic) reveals that having `CUDA::cudart_static` as a `PUBLIC` dependency will affect the downstream user.

Error
```
[1/1] Linking C executable api-demo
FAILED: api-demo
: && /usr/bin/cc  -rdynamic CMakeFiles/api-demo.dir/c-api-demo.o -o api-demo -L/usr/local/lib64 -Wl,-rpath,/usr/local/lib64  /usr/local/lib64/libxgboost.so  -lCUDA::cudart_static && :
/usr/bin/ld: cannot find -lCUDA::cudart_static
```

I considered revising the CMake config to add `find_dependency(CUDAToolkit)`. It fixed the build, but added an unneeded static lib to the downstream lib:
```
[2/2] : && /usr/bin/cc  -rdynamic CMakeFiles/api-demo.dir/c-api-demo.o -o api-demo -L/usr/local/lib64   -L/usr/local/cuda-11.8/targets/x86_64-linux/lib -Wl,-rpath,/usr/local/lib64  /usr/local/lib64/libxgboost.so  /usr/local/cuda-11.8/targets/x86_64-linux/lib/libcudart_static.a  -lpthread  -ldl  -lrt && :
```

So I think we should make `CUDA::cudart_static` a `PRIVATE` dependency.

Reply:
I believe @hcho3 is right here.

> maybe downstream users need to install CUDA runtime too? Let me take another look.

No, XGBoost runs on CPU-only machines even if compiled with CUDA. We statically link a CUDA runtime (cudart) shim into XGBoost shared object, the user should not need CUDA dependency.

Reply:
Ok agreed. Thanks for looking into this! 🙏